### PR TITLE
Some cleanups targeting final role migration

### PR DESCRIPTION
**Pull request by danieljames-dj**

### CODE REVIEW

File: app/jobs/sync_mailing_lists_job.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class SyncMailingListsJob < WcaCronjob
   end
 
   def perform
-    GsuiteMailingLists.sync_group("leaders@worldcubeassociation.org", TeamMember.current.in_official_team.leader.map(&:user).map(&:email))
+    GsuiteMailingLists.sync_group("leaders@worldcubeassociation.org", UserGroup.teams_committees.map(&:lead_user).compact.map(&:email))
```
Comment (gregorbg) [inline app/jobs/sync_mailing_lists_job.rb:10]: Does this account for the **current** lead users only?

File: app/models/user.rb [hunk 2]
```diff
@@ -65,9 +65,9 @@ def self.eligible_voters
   end
 
   def self.leader_senior_voters
-    team_leaders = TeamMember.current.in_official_team.leader.map(&:user)
+    team_leaders = UserGroup.teams_committees.map(&:lead_user)
```
Comment (gregorbg) [inline app/models/user.rb:68]: Same question as above (feel free to resolve both by just answering one): Does this account for **current** Team Leaders only?

File: spec/features/eligible_voters_spec.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@
       user.team_members.find_by_team_id(wrc_team_id).update!(end_date: 1.day.ago)
     end
   }
-  let!(:team_leader) { FactoryBot.create(:user, :wrc_member, team_leader: true) }
+  let!(:team_leader) { FactoryBot.create(:user, :wrt_member, team_leader: true) }
```
Comment (gregorbg) [inline spec/features/eligible_voters_spec.rb:18]: Why? This change seems quite arbitrary...

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -200,7 +200,7 @@
       # WST is not moderator by default, admin status is granted manually in
       # Discourse
       expect(answer_sso.moderator).to be false
-      expect(answer_sso.add_groups).to eq "wst,delegate"
+      expect(answer_sso.add_groups).to eq "delegate,wst"
```
Comment (gregorbg) [inline spec/requests/users_spec.rb:203]: Might be cool to change the SSO interface to make sure the list of string tags (`wst`, `delegate`, etc.) is always sorted alohabetically. That way, we have a defined behavior of tests and don't need to worry about test chokes if some order of the rows in the DB changes.

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: spec/requests/users_spec.rb (hunk 4 above)
Comment (danieljames-dj) [reply spec/requests/users_spec.rb:203]: Okay done

**Review by gregorbg — APPROVED** [review]

**Pull request merged**